File: tests/fixtures/textmate/java/basic.java
package fixtures.java;

import java.util.List;

@Deprecated
final class Basic<T extends Number> {
    private final List<T> values;

    Basic(List<T> values) {
        this.values = List.copyOf(values);
    }

    /* Multiline state with café and λ;
       astral glyphs 🚀 and 𝌆 are included and closed. */
    String render(int count) {
        String document = """
                café λ launches 🚀
                rare symbol: 𝌆
                """;
        return switch (count) {
            case 0 -> "empty";
            default -> "%s (%d): %s".formatted(document.strip(), count, values);
        };
    }
}

File: tests/fixtures/textmate/csharp/basic.cs
using System;
using System.Collections.Generic;

namespace Fixtures.CSharp;

public sealed record Launch(string Name, IReadOnlyList<int> Codes)
{
    /* Multiline state:
       café, λ, 🚀, and 𝌆 stay inside a closed comment. */
    public string Describe(int count)
    {
        var path = @"C:\café\λ\launch.txt";
        var message = $"{Name} sends 🚀 number {count:D2}";
        var document = $$"""
            {
              "symbol": "𝌆",
              "message": "{{message}}"
            }
            """;
        return count > 0 ? $"{path}: {document}" : "idle";
    }
}
